includes/MirahezeMagicHooks.php

```diff
@@ -254,4 +254,23 @@ public static function onSkinAddFooterLinks( Skin $skin, string $key, array &$fo
 			$footerItems['donate'] = $skin->footerLink( 'miraheze-donate', 'miraheze-donatepage' );
 		}
 	}
+
+	public static function onUserGetRightsRemove( User $user, array &$aRights ) {
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'mirahezemagic' );
+		// Remove read from stewards on staff wiki.
+		if ( $config->get( 'DBname' ) === 'staffwiki' && $user->isLoggedIn() &&
+		    class_exists( 'CentralAuthUser' ) ) {
+			$centralAuthUser = CentralAuthUser::getInstance( $user );
+			if ( $centralUser->exists() && $centralUser->isAttached() &&
+			    isset( $centralAuthUser->getGlobalGroups()['steward'] ) ) {
+				foreach ( $aRights as $i => $right ) {
```

Comment:
Why not just $aRights['read'] ?? unset or similar?

Reply:
Changed the logic thus that is unneeded.